Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm

```diff
@@ -1302,6 +1302,59 @@ HTTPServer server({
     EXPECT_WK_STREQ("mousedown,40,40", eventTypes[1]);
     EXPECT_WK_STREQ("mouseup,40,40", eventTypes[2]);
 }
+
+TEST(SiteIsolation, DragEvents)
+{
+    auto mainframeHTML = "<script>"
+    "    window.events = [];"
+    "    addEventListener('message', function(event) {"
+    "        window.events.push(event.data);"
+    "    });"
+    "</script>"
+    "<iframe width='300' height='300' src='https://domain2.com/subframe'></iframe>"_s;
+
+    auto subframeHTML = "<body>"
+    "<div id='draggable' draggable='true' style='width: 100px; height: 100px; background-color: blue;'></div>"
+    "<script>"
+    "    draggable.addEventListener('dragstart', window.parent.postMessage('dragstart', '*'));"
+    "    draggable.addEventListener('drag', window.parent.postMessage('drag', '*'));"
+    "    draggable.addEventListener('dragend', window.parent.postMessage('dragend', '*'));"
+    "    draggable.addEventListener('dragenter', window.parent.postMessage('dragenter', '*'));"
+    "    draggable.addEventListener('dragleave', window.parent.postMessage('dragleave', '*'));"
+    "</script>"
+    "</body>"_s;
+
+    HTTPServer server({
+        { "/mainframe"_s, { mainframeHTML } },
+        { "/subframe"_s, { subframeHTML } }
+    }, HTTPServer::Protocol::HttpsProxy);
+
+    auto navigationDelegate = adoptNS([TestNavigationDelegate new]);
+    [navigationDelegate allowAnyTLSCertificate];
+    auto configuration = server.httpsProxyConfiguration();
+    enableSiteIsolation(configuration);
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:configuration]);
+    webView.get().navigationDelegate = navigationDelegate.get();
+
+    [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"https://domain1.com/mainframe"]]];
+    [navigationDelegate waitForDidFinishNavigation];
+
+    CGPoint beginPoint = [webView convertPoint:CGPointMake(50, 50) toView:nil];
+    CGPoint endPoint = [webView convertPoint:CGPointMake(150, 150) toView:nil];
+    [webView mouseEnterAtPoint:beginPoint];
+    [webView mouseDownAtPoint:beginPoint simulatePressure:NO];
+    [webView mouseMoveToPoint:endPoint withFlags:0];
+    [webView mouseUpAtPoint:endPoint];
+    [webView waitForPendingMouseEvents];
```

Comment:
I wonder if it would be easier to use `DragAndDropSimulator` here (similar to the other tests in `DragAndDropTestsMac.mm`). It looks like you need a custom web view configuration here, but this constructor should do the trick:

```
- (instancetype)initWithWebViewFrame:(CGRect)frame configuration:(WKWebViewConfiguration *)configuration;
```

You can then do:

```
[simulator runFrom:CGPointMake(50, 50) to:CGPointMake(150, 150)];
```

…in case this doesn't Just Work™, I'd say adopting `DragAndDropSimulator` definitely doesn't block landing; however, it might mean that many of the other existing drag and drop tests could be failing with site isolation enabled, so we'll need to fix those eventually.